homeassistant/components/zwave_js/addon.py

```diff
@@ -0,0 +1,246 @@
+
+    async def async_is_addon_running(self) -> bool:
+        """Return True if Z-Wave JS add-on is running."""
+        addon_info = await self.async_get_addon_info()
+        return bool(addon_info["state"] == "started")
+
+    async def async_is_addon_installed(self) -> bool:
+        """Return True if Z-Wave JS add-on is installed."""
+        addon_info = await self.async_get_addon_info()
+        return addon_info["version"] is not None
+
+    async def async_get_addon_options(self) -> dict:
+        """Get Z-Wave JS add-on options."""
+        addon_info = await self.async_get_addon_info()
+        return cast(dict, addon_info["options"])
+
+    @api_error("Failed to set the Z-Wave JS add-on options")
+    async def async_set_addon_options(self, config: dict) -> None:
+        """Set Z-Wave JS add-on options."""
+        options = {"options": config}
+        await async_set_addon_options(self._hass, ADDON_SLUG, options)
+
+    @api_error("Failed to install the Z-Wave JS add-on")
+    async def async_install_addon(self) -> None:
+        """Install the Z-Wave JS add-on."""
+        await async_install_addon(self._hass, ADDON_SLUG)
+
+    @callback
+    def async_schedule_install_addon(
+        self, usb_path: str, network_key: str
+    ) -> asyncio.Task:
+        """Schedule a task that installs and sets up the Z-Wave JS add-on.
+
+        Only schedule a new install task if the there's no running task.
+        """
+        if not self._install_task or self._install_task.done():
+            LOGGER.warning("Z-Wave JS add-on is not installed. Installing add-on")
+            self._install_task = self._async_schedule_addon_operation(
+                self.async_install_addon,
+                partial(self.async_setup_addon, usb_path, network_key),
+            )
+        return self._install_task
+
+    @api_error("Failed to uninstall the Z-Wave JS add-on")
+    async def async_uninstall_addon(self) -> None:
+        """Uninstall the Z-Wave JS add-on."""
+        await async_uninstall_addon(self._hass, ADDON_SLUG)
+
+    @api_error("Failed to update the Z-Wave JS add-on")
+    async def async_update_addon(self) -> None:
+        """Update the Z-Wave JS add-on if needed."""
+        addon_info = await self.async_get_addon_info()
+        addon_version = addon_info["version"]
+        update_available = addon_info["update_available"]
+
+        if addon_version is None:
+            raise AddonError("Z-Wave JS add-on is not installed")
+
+        if not update_available:
+            return
+
+        await async_update_addon(self._hass, ADDON_SLUG)
+
+    @callback
+    def async_schedule_update_addon(self) -> asyncio.Task:
+        """Schedule a task that updates and sets up the Z-Wave JS add-on.
+
+        Only schedule a new update task if the there's no running task.
+        """
+        if not self._update_task or self._update_task.done():
+            LOGGER.warning("Trying to update the Z-Wave JS add-on")
```

Comment:
```suggestion
            LOGGER.debug("Trying to update the Z-Wave JS add-on")
```
Does not seem important enough to warn about

Reply:
I wanted to let the user know that we're trying to resolve the error situation:
```
2021-03-02 21:08:14 ERROR (MainThread) [custom_components.zwave_js] Invalid server version: Z-Wave JS Server version is incompatible: 1.0.0-beta.6 a version is required that supports at least api schema 1
2021-03-02 21:08:14 WARNING (MainThread) [custom_components.zwave_js] Trying to update the Z-Wave JS add-on
```

We could limit it to info level, but I think debug is a bit too invisible for users.

Reply:
Info works